custommutator/mutators/extended_key.h

```diff
@@ -0,0 +1,101 @@
+#ifndef BITCOINFUZZ_CUSTOMMUTATOR_EXTENDED_KEY_H
+#define BITCOINFUZZ_CUSTOMMUTATOR_EXTENDED_KEY_H
+
+#include <custommutator/utils/base58.h>
+// Custom mutator for Base58 encoded extended keys (xpub/xprv).
+// so far only adds the "xpub" prefix and converts random data to base58
+// BASE 58 lib to be added?
+// add base58.h like this? 
+// #include <modules/custommutator/base58.h>
+// or borrow from bitcoin core? 
+// #include <modules/bitcoin/base58.h>
+// or custom implement base58 encode here?
+// note that base58.h requires linking other bitcoin core files (s)
+inline constexpr size_t EXTKEY_CHAR_SIZE_WITHOUT_XPUB = 107;
+inline constexpr size_t BIP32_PAYLOAD_LEN             = 78;
+inline constexpr size_t VERSION_PREFIX_LEN            = 4;
+
+extern "C" size_t LLVMFuzzerMutate(uint8_t *Data, size_t Size, size_t MaxSize);
+
+extern "C" size_t LLVMFuzzerCustomMutator(uint8_t *fuzz_data, size_t size, size_t max_size,
+                               unsigned int seed) {
+    // 1) Decide key type via env
+    std::string selected_key_type = "xpub"; // default
+                                
+    if (const char* env = std::getenv("EXTKEYTYPE")) {
+        std::string e(env);
+        if (e == "xpub" || e == "xprv" || e == "tpub" || e == "tprv") {
+            selected_key_type = e;
+        }
+    }
+    //printf("Selected EXTKEYTYPE: %s\n", selected_key_type.c_str());
+    // xpub/xprv bytes
+    const uint8_t XPUB_VERSION[4] = {0x04, 0x88, 0xB2, 0x1E}; // xpub
+    const uint8_t XPRV_VERSION[4] = {0x04, 0x88, 0xAD, 0xE4}; // xprv
+    const uint8_t TPUB_VERSION[4] = {0x04, 0x35, 0x87, 0xCF}; // tpub
+    const uint8_t TPRV_VERSION[4] = {0x04, 0x35, 0x83, 0x94}; // tprv
+
+    std::string input_str(reinterpret_cast<char*>(fuzz_data), size);
+
+    //decode base58check to raw bytes
+    std::vector<unsigned char> decoded;
+    bool ok = DecodeBase58Check(input_str, decoded, /*max_ret_len=*/(int)max_size);
+
+    // if decode failed, use raw input bytes
+    if (!ok) {
+        decoded.assign(input_str.begin(), input_str.end());
+    }
+
+    // pad or truncate to 78 bytes
+    decoded.resize(BIP32_PAYLOAD_LEN, 0x00);
+
+    std::vector<uint8_t> scratch(max_size);
+    // copy the 78-byte payload into scratch[0..77]
+    std::memcpy(scratch.data(), decoded.data(), BIP32_PAYLOAD_LEN);
+
+
+    size_t inner_offset = VERSION_PREFIX_LEN;
+    size_t inner_size = BIP32_PAYLOAD_LEN - VERSION_PREFIX_LEN; 
+
+    // mutate everything
+    size_t returned_size = LLVMFuzzerMutate(
+        scratch.data(),
+        BIP32_PAYLOAD_LEN,
+        max_size
+    );
+    if (returned_size > BIP32_PAYLOAD_LEN) returned_size = BIP32_PAYLOAD_LEN;
+
+    // pad remaining bytes if needed
+    if (returned_size < BIP32_PAYLOAD_LEN) {
+        std::memset(scratch.data() + returned_size, 0, BIP32_PAYLOAD_LEN - returned_size);
```

Comment:
The returned size should be set to BIP32_PAYLOAD_LEN after adding pad to scratch.


```suggestion
    if (returned_size < BIP32_PAYLOAD_LEN) {
        std::memset(scratch.data() + returned_size, 0, BIP32_PAYLOAD_LEN - returned_size);
        returned_size = BIP32_PAYLOAD_LEN;
```